framework/include/mesh/MooseMesh.h

```diff
@@ -607,6 +607,8 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
    */
   void setSubdomainName(SubdomainID subdomain_id, SubdomainName name);
 
+  static void setSubdomainName(MeshBase & mesh, SubdomainID subdomain_id, SubdomainName name);
```

Comment:
A doc string would be helpful for this method. It's not exactly clear what to do with the arguments just based on the name. I can tell what it does when I look at the implementation. Also you can pass your string by const ref.